core/src/main/java/org/elasticsearch/search/sort/SortBuilder.java

```diff
@@ -69,6 +78,132 @@
         PARSERS = unmodifiableMap(parsers);
     }
 
+    public static class NestedSort implements Writeable, Reader<NestedSort>, ToXContentObject {
```

Comment:
Even though it is not strictly a sort builder, can you rename it NestedSortBuilder for consistency and move it to its own class?